src/Storages/IMessageProducer.cpp

```diff
@@ -12,7 +12,16 @@ void AsynchronousMessageProducer::start(const ContextPtr & context)
 {
     LOG_TEST(log, "Executing startup");
 
-    initialize();
+    try
+    {
+        initialize();
+    }
+    catch (...)
+    {
+        finished = true;
+        throw;
+    }
```

Comment:
Previously if we tried to insert into RabbitMQ/NATS that is not initialized, it led to deadlock in `AsynchronousMessageProducer::finish` on `schedule.wait()`. This fixes it